sites/all/modules/custom/jdt_user_notes/jdt_user_notes.module

```diff
@@ -103,10 +92,10 @@ function jdt_user_notes_add_submit($form, &$form_state) {
  * Custom handler for jdt_user_notes_edit().
  */
 function jdt_user_notes_edit($note_id, $article_nid) {
-  $check = jdt_user_notes_read($note_id);
+  $existing_note = jdt_user_notes_read($note_id);
 
-  if (empty($check) || $check['article_nid'] != $article_nid) {
-    $message = 'This article note does not exist!';
+  if (empty($existing_note) || $existing_note['article_nid'] != $article_nid) {
+    $message = 'This article note does not exist.';
     $form = drupal_get_form('jdt_user_notes_error_form', $message);
```

Comment:
There are many options to go about this, you can look into this when you have time, not required for now:
* Custom page access callback to check if note exists before rendering the form.
* Or, redirect back with an error message so you don't need to have an extra custom form like jdt_user_notes_error_form